### PR TITLE
Generate stable machine-id and DHCP client ID on POWER KVM

### PR DESCRIPTION
**Pull request by xnox**

### CODE REVIEW

**Comment by poettering** [comment]
(Ideally we'd have a single helper function which would query the IDs at both places)

**Comment by xnox** [comment]
Ok, note I'll be working on the 3rd ID for s390x too shortly.

Then will refactor to do them all.

s390x is a bit tricky as there is stable UUID for both LPAR (aka none) and KVM VMs.

It's also interesting that we do not initialize machine-id with these stable ids. (only in virtualization kvm, but not in bare metal). I guess the logic there is that when you reinstall, one wants a "fresh" machine.

File: src/core/machine-id-setup.c [hunk 1]
```diff
@@ -68,6 +68,11 @@ static int generate_machine_id(const char *root, sd_id128_t *ret) {
                                 log_info("Initializing machine ID from KVM UUID.");
                                 return 0;
                         }
+                        /* on POWER, it's exported here instead */
+                        if (id128_read("/sys/firmware/devicetree/base/vm,uuid", ID128_UUID, ret) >= 0) {
+                                log_info("Initializing machine ID from KVM UUID.");
```
Comment (keszybz) [inline src/core/machine-id-setup.c:73]: This should be differentiated from the previous call, maybe
`"Initializing machine ID from KVM UUID in /sys/class/dmi/id/product_uuid."`
`"Initializing machine ID from KVM UUID in /sys/firmware/devicetree/base/vm,uuid."`

File: src/hostname/hostnamed.c [hunk 2]
```diff
@@ -116,6 +116,8 @@ static int context_read_data(Context *c) {
                 return r;
 
         r = id128_read("/sys/class/dmi/id/product_uuid", ID128_UUID, &c->uuid);
+        if (r < 0)
```
Comment (keszybz) [inline src/hostname/hostnamed.c]: `r == -ENOENT` ?

**Comment by poettering** [comment]
ping?

File: src/core/machine-id-setup.c (hunk 1 above)
Comment (xnox) [reply src/core/machine-id-setup.c:73]: The message will only be printed once. And I didn't want to change it and has the same meaning irrespective of where it came from: `dmi` or `devicetree` paths. As it really is the KVM UUID that KVM has setup and passed through. And only one or the other path exist on any given architecture. Personally i would even `#ifdef` the code, but that's not the systemd source code style.

File: src/hostname/hostnamed.c (hunk 2 above)
Comment (xnox) [reply src/hostname/hostnamed.c]: yes

**Comment by xnox** [comment]
@keszybz fixed `-ENOENT` & trying to justify using the same info string. Let me know if you reject my justification for using architecture agnostic info string =)

File: src/core/machine-id-setup.c (hunk 1 above)
Comment (keszybz) [reply src/core/machine-id-setup.c:73]: OK.

**Pull request merged**